src/v/cluster/cluster_link/frontend.cc

```diff
@@ -492,14 +492,50 @@ errc frontend::validator::validate_mutation(const cluster_link_cmd& cmd) const {
           if (!meta.has_value()) {
               return errc::does_not_exist;
           }
-          return errc::success;
+          const auto& md = meta->get();
+          const auto is_removable =
+            [](const ::cluster_link::model::mirror_topic_status s) {
+                switch (s) {
+                case ::cluster_link::model::mirror_topic_status::active:
+                case ::cluster_link::model::mirror_topic_status::paused:
+                case ::cluster_link::model::mirror_topic_status::failing_over:
+                case ::cluster_link::model::mirror_topic_status::promoting:
+                    return false;
+                case ::cluster_link::model::mirror_topic_status::promoted:
+                case ::cluster_link::model::mirror_topic_status::failed_over:
+                case ::cluster_link::model::mirror_topic_status::failed:
+                    return true;
+                }
+            };
+          const auto mirror_topic_states
+            = md.state.mirror_topics | std::views::values
+              | std::views::transform(
+                &::cluster_link::model::mirror_topic_metadata::status);
+
+          if (
+            mirror_topic_states.empty()
+            || std::ranges::any_of(mirror_topic_states, is_removable)) {
```

Comment:
should this be `all_of` ?

If i understand that correctly we can only remove link if all topics are removable

Reply:
oh duh, you are right, I had it correct originally but filpped it by mistake because a test was failing. will add test coverage.